R/pb_read.R

```diff
@@ -0,0 +1,90 @@
+#' Read one file into memory
+#'
+#' A convenience wrapper around writing an object to a temporary file and then
+#' uploading to a specified repo/release. This convenience comes at a cost to
+#' performance efficiency, since it first downloads the data to disk and then
+#' reads the data from disk into memory. See `vignette("cloud_native")` for
+#' alternative ways to bypass this flow and work with the data directly.
+#'
+#' @param file string: file name
+#' @param repo string: GH repository name in format "owner/repo". Default
+#' `guess_repo()` tries to guess based on current working directory's git repo
+#' @param tag  string: tag for the GH release, defaults to "latest"
+#' @param read_function function: used to read in the data, where the file is
+#' passed as the first argument and any additional arguments are subsequently
+#' passed in via `...`. Default `guess_read_function(file)` will check the file
+#' extension and try to find an appropriate read function if the extension is one
+#' of rds, csv, tsv, parquet, txt, or json, and will abort if not found.
+#' @param ... additional arguments passed to `read_function` after file
+#' @param .token GitHub authentication token, see [gh::gh_token()]
```

Comment:
Fleshed out more of this section, explaining how the read_function will be used (file as first arg etc)